### DISCUSSION TITLE
Typing next steps

### DISCUSSION THREAD

**Discussion by Yobmod** [gitpython-developers/GitPython#1313] [Ideas]
- [x] Drop 3.6 & add 3.10
- [x] Final 3.6 release
- [x] Add CI tests for early patch versions (e.g. py3.7.1)
- [ ] Add `__future__.annotations` imports
- [ ] Replace Optional and Union with |  (pep604)
- [ ] Re-add TypeGuards
- [ ] Replace more casts with TypeGuards
- [ ] Check runtime types of TBDs (BearType, TypeGuard, ~StrongTyping~)
- [ ] Add more TypedDicts and/or Dataclasses
- [ ] Fix `no-implicit reexport` errors
- [ ] Fix other `--strict` errors

**Comment by Byron** [comment]
Thanks for all your great work and taking typing even further! Please feel free to ping me whenever you need a release, and if something is particularly important you are welcome to add that to `changes.rst` as well. I might miss a lot and definitely won't be reproducing finer details due to a lack of awareness.
